src/contact.rs

```diff
@@ -1370,11 +1359,48 @@ fn cat_fingerprint(
 impl Context {
     /// determine whether the specified addr maps to the/a self addr
     pub async fn is_self_addr(&self, addr: &str) -> Result<bool> {
-        if let Some(self_addr) = self.get_config(Config::ConfiguredAddr).await? {
-            Ok(addr_cmp(self_addr, addr))
-        } else {
-            Ok(false)
+        let addr = addr_normalize(addr).to_lowercase();
+        // Addresses in `self_addrs` are already normalized and lowercase
+        self.sql
+            .exists(
+                "SELECT COUNT(*) FROM self_addrs WHERE addr=?;",
+                paramsv![addr],
+            )
+            .await
+    }
+
+    pub async fn add_self_addr(&self, addr: &str) -> Result<()> {
```

Comment:
will `addr`  become the new primary implicitely?  If so, the doc string should say so. If not, then i guess the caller is responsible for also setting ConfiguredAddr when adding a new primary address (the default use case for aeap), right?